src/test/java/nl/knaw/dans/layerstore/LayerManagerGetLayerTest.java

```diff
@@ -15,5 +15,120 @@
  */
 package nl.knaw.dans.layerstore;
 
-public class LayerManagerGetLayerTest {
+import org.assertj.core.api.AssertionsForClassTypes;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.nio.file.Files;
+
+import static java.lang.Thread.sleep;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class LayerManagerGetLayerTest extends AbstractTestWithTestDir {
+
+    @Test
+    public void should_throw_when_layer_id_does_not_exist() throws IOException {
+        // Given
+        var layerManager = new LayerManagerImpl(stagingDir, new ZipArchiveProvider(archiveDir));
+
+        // When
+        assertThatThrownBy(() -> layerManager.getLayer(1234567890123L))
+            // Then
+            .isInstanceOf(IllegalArgumentException.class)
+            .hasMessageContaining("No layer found with id 1234567890123");
+
+    }
+
+    @Test
+    public void should_find_garbage_in_stagingDir_created_after_creating_LayerManagerImpl_object() throws IOException {
+        // Given
+        var layerManager = new LayerManagerImpl(stagingDir, new ZipArchiveProvider(archiveDir));
+        assertThat(layerManager.getTopLayer().getId()).isNotEqualTo(0L);
+        Files.createDirectories(stagingDir.resolve("0"));
+
+        //When
+        var layer = layerManager.getLayer(0L);
+
+        // Then
+        assertThat(layer.getId()).isNotEqualTo(layerManager.getTopLayer().getId());
```

Comment:
I also don't understand the point of the test.

Reply:
It shows inconsistent behavior between the constructor and getLayer.